### PR TITLE
feature(sct-builders): move builder to auto scaling group

### PR DESCRIPTION
**Pull request by fruch**
Since we are not using those builder all the time, we are starting to create them as each time we start a job.

* very small machine that are only use to spin off the test
* in some cases it should be able to run the test (artifact test or SCT CI)
* this is creating the ASGs and launch templates in all regions we operate
* configure those ASGs in jenkins
* switch all monitors to ubuntu jammy, since the old centos images
  we use aren't available anymore across all regions

Testing:
- [x] CI - both integration test / docker backend
- [x] longevity - https://jenkins.scylladb.com/job/scylla-staging/job/fruch/job/longevity-100gb-4h-test/3/
- [x] longevity - run on new region `ca-central-1` - https://jenkins.scylladb.com/job/scylla-staging/job/fruch/job/longevity-100gb-4h-test/10/
- [x] test artifact test - docker - https://jenkins.scylladb.com/job/scylla-staging/job/fruch/job/artifacts-docker-test/17/
- [x] show monitor job - test and switch to same as CI
- [x] cloud report - 
- [ ] azure -
- [ ] azure artifact test -

## PR pre-checks (self review)

- [ ] I followed [KISS principle](https://en.wikipedia.org/wiki/KISS_principle) and [best practices](https://docs.google.com/document/d/1jihgOKb5iGRlD8_HQ92O0JbLk1kASUoZT23i_MXFSKI)
- [ ] I didn't leave commented-out/debugging code
- [ ] I added the relevant `backport` labels
- [ ] New configuration option are added and documented (in `sdcm/sct_config.py`)
- [ ] I have added tests to cover my changes (Infrastructure only - under `unit-test/` folder)
- [ ] All new and existing unit tests passed (CI)
- [ ] I have updated the Readme/doc folder accordingly (if needed)

### CODE REVIEW

**Comment by fruch** [comment]
Switch all monitor images to Ubuntu

```python
import boto3

def find_latest_canonical_ubuntu_ami(region):
    # Create a boto3 client for EC2
    ec2_client = boto3.client('ec2', region_name=region)

    # Use the describe_images method to search for Ubuntu AMIs
    response = ec2_client.describe_images(
        Filters=[
            {'Name': 'name', 'Values': ['ubuntu/images/hvm-ssd/ubuntu-*-20*']},
            {'Name': 'architecture', 'Values': ['x86_64']},
            {'Name': 'virtualization-type', 'Values': ['hvm']},
            {'Name': 'root-device-type', 'Values': ['ebs']},
            {'Name': 'owner-alias', 'Values': ['canonical']}
        ]
    )

    # Sort the AMIs by creation date
    amis = sorted(response['Images'], key=lambda x: x['CreationDate'])

    # Return the latest AMI
    return amis[-1]

# Example usage
latest_ami = find_latest_canonical_ubuntu_ami('us-east-1')
print(latest_ami)


```

File: sct.py [hunk 1]
```diff
@@ -987,6 +987,15 @@ def unit_tests(test):
 @click.option("-t", "--test", required=False, default="",
               help="Run specific test file from unit-tests directory")
 def integration_tests(test):
+    get_test_config().logdir()
+    add_file_logger()
+    local_cluster = mini_k8s.LocalKindCluster(
```
Comment (soyacz) [inline sct.py:992]: why do we need it?

**Review by soyacz — DISMISSED** [review]
quite complicated change, hard to find issues for me, just a few questions.

File: Jenkinsfile [hunk 2]
```diff
@@ -154,13 +154,37 @@ pipeline {
...
+                                    }
+                                }
+                                sh """#!/bin/bash
+                                    set -xe
+                                    echo "start integration-tests ..."
+                                    RUNNER_IP=\$(cat sct_runner_ip||echo "")
+                                    if [[ -n "\${RUNNER_IP}" ]] ; then
+                                        ./docker/env/hydra.sh --execute-on-runner \${RUNNER_IP} integration-tests
+                                    else
+                                        ./docker/env/hydra.sh integration-tests
```
[the remark this answers is not in the document]
Comment (fruch) [reply Jenkinsfile:181]: we would actually never going to get here, if creation of runner would fail.

yeah this part can be remove, it's just me copying the logic from other steps

File: sct.py (hunk 1 above)
Comment (fruch) [reply sct.py:992]: since we are taking a bare SCT runner, some of the system configuration that is done for local kind, we need it as well for the integration tests, yes I probability could add a comment here stating why we are doing this.

File: Jenkinsfile (hunk 2 above)
[the remark this answers is not in the document]
Comment (fruch) [reply Jenkinsfile]: removed the if

File: sct.py (hunk 1 above)
Comment (fruch) [reply sct.py:996]: Added a comment, saying exactly that

File: Jenkinsfile [hunk 3]
```diff
@@ -66,7 +66,7 @@ def runRestoreMonitoringStack(){
 pipeline {
     agent {
         label {
-            label "sct-builders"
+            label "aws-sct-builders-eu-west-1-v2-CI"
```
Comment (roydahan) [inline Jenkinsfile:69]: Maybe a more meaningful name?
When I back to these names later in Jenkins, I don't remember what they means and if they are the correct ones and need to check the code.

File: defaults/aws_config.yaml [hunk 4]
```diff
@@ -8,24 +8,54 @@ instance_type_loader: 'c5.xlarge'
...
+    ami_id_loader: 'ami-06037dd084892cb6e' # Loader dedicated AMI scylla-qa-loader-ami-v19
+    ami_id_monitor: 'ami-07ac54771249f286c' # ubuntu/images/hvm-ssd/ubuntu-jammy-22.04-amd64-server-20221206 Canonical, Ubuntu, 22.04 LTS, amd64 jammy image build on 2022-12-06
+  ca-central-1: # Canada (Central)
+    ami_id_loader: 'ami-0f4f26c6055d001d9' # Loader dedicated AMI scylla-qa-loader-ami-v19
+    ami_id_monitor: 'ami-09bbc74353976b63f' # ubuntu/images/hvm-ssd/ubuntu-jammy-22.04-amd64-server-20221206 Canonical, Ubuntu, 22.04 LTS, amd64 jammy image build on 2022-12-06
+  ap-southeast-1: # Asia Pacific (Singapore)
+    ami_id_loader: 'ami-00a2cba01e3b307c1' # Loader dedicated AMI scylla-qa-loader-ami-v19
+    ami_id_monitor: 'ami-029562ad87fe1185c' # ubuntu/images/hvm-ssd/ubuntu-jammy-22.04-amd64-server-20221206 Canonical, Ubuntu, 22.04 LTS, amd64 jammy image build on 2022-12-06
+  ap-southeast-2: # Asia Pacific (Sydney)
+    ami_id_loader: 'ami-0679f687c2ab1a829' # Loader dedicated AMI scylla-qa-loader-ami-v19
```
Comment (roydahan) [inline defaults/aws_config.yaml:42]: Why do we need all these regions?
We probably don't have quota there.

File: sdcm/utils/aws_builder.py [hunk 5]
```diff
@@ -26,12 +29,81 @@
...
+import hudson.model.*
+import com.amazon.jenkins.ec2fleet.EC2FleetCloud
+import jenkins.model.Jenkins
+
+// just modify this config other code just logic
+config = [
+    name: "{name}",
+    region: "{region_name}",
+    fleetId: "{asg_id}",
+    idleMinutes: 10,
```
Comment (roydahan) [inline sdcm/utils/aws_builder.py:45]: Is it the idle time before scale down?
Maybe we should consider larger time?

File: sdcm/utils/aws_builder.py [hunk 6]
```diff
@@ -26,12 +29,81 @@
...
+    minSize: {min_size},
+    maxSize: {max_size},
+    numExecutors: 1,
```
Comment (roydahan) [inline sdcm/utils/aws_builder.py:48]: do we want 1 to 1 mapping? wouldn't we have too many builder in this case?

File: test-cases/PR-provision-test-docker.yaml [hunk 7]
```diff
@@ -7,6 +7,8 @@ n_loaders: 1
 n_monitor_nodes: 1
 n_db_nodes: 1
 
+instance_type_runner: c5.2xlarge
```
Comment (roydahan) [inline test-cases/PR-provision-test-docker.yaml:10]: why not something much smaller like c5.large?

File: utils/copy_ami_to_all_regions.sh [hunk 8]
```diff
@@ -0,0 +1,10 @@
+AMI_ID=ami-00c1a16b5136fbb7c
+AMI_NAME='scylla-qa-loader-ami-v19'
+SOURCE_REGION=us-east-1
+TARGET_REGIONS='eu-north-1 ap-south-1 eu-west-3 eu-west-2 eu-west-1 ap-northeast-2 ap-northeast-1 sa-east-1 ca-central-1 ap-southeast-1 ap-southeast-2 eu-central-1 us-east-2 us-west-1 us-west-2'
```
Comment (roydahan) [inline utils/copy_ami_to_all_regions.sh:4]: again, not sure why we need all these regions.

File: Jenkinsfile (hunk 3 above)
Comment (fruch) [reply Jenkinsfile:69]: the name now tell you it's in aws, which region it is, and that it's used for CI.
what more meaning you are missing here ?

File: defaults/aws_config.yaml (hunk 4 above)
Comment (fruch) [reply defaults/aws_config.yaml:42]: I was under the impression we do want ability to run on any region we want.
and that we limited it just to the region we tested so far. (and by having running Jenkins workers)

You want to leave the limit in place ? to only the 6-7 region that was there before ?

File: sdcm/utils/aws_builder.py (hunk 6 above)
Comment (fruch) [reply sdcm/utils/aws_builder.py:48]: I can play with this number, but I'm not sure how high I can really set it. 
we can start with 30 and see how it's works

File: sdcm/utils/aws_builder.py (hunk 5 above)
Comment (fruch) [reply sdcm/utils/aws_builder.py:45]: why ? it would be killed and a new one would be created.
what do we loose if something like this happens ? we wait 1-2 minutes for a new one to spin up.

File: test-cases/PR-provision-test-docker.yaml (hunk 7 above)
Comment (fruch) [reply test-cases/PR-provision-test-docker.yaml:10]: 2 cpus isn't enough to running the docker backend (also the integration test) reliably.

Comment (fruch) [reply test-cases/PR-provision-test-docker.yaml:10]: those are optional CI phase that take ~30min each

File: defaults/aws_config.yaml (hunk 4 above)
Comment (roydahan) [reply defaults/aws_config.yaml:42]: I think the general approach now is to limit to less regions, so we won't have too many AMIs and snapshots that we will pay for.

File: sdcm/utils/aws_builder.py (hunk 5 above)
Comment (roydahan) [reply sdcm/utils/aws_builder.py:45]: Yes, I guess.
Hard to tell without seeing it working.

File: sdcm/utils/aws_builder.py (hunk 6 above)
Comment (roydahan) [reply sdcm/utils/aws_builder.py:48]: Do you mean 30 as the maxSize?
I'm sure we need more than that at some points.

File: test-cases/PR-provision-test-docker.yaml (hunk 7 above)
Comment (roydahan) [reply test-cases/PR-provision-test-docker.yaml:10]: oh it's running locally on this machine as docker backend.
I agree.

File: sdcm/utils/aws_builder.py (hunk 6 above)
Comment (fruch) [reply sdcm/utils/aws_builder.py:48]: No, 30 executers on each instance,

The problem is that artifacts might accumulate and I'm not sure 30 artifacts tests running at the same time on 2cpu would get very far...

File: sdcm/utils/aws_builder.py [hunk 9]
```diff
@@ -137,18 +209,144 @@ def associate_elastic_ip(self):
...
+                                                     }]
+                                                 )
+
+        except botocore.exceptions.ClientError as error:
+            LOGGER.debug(error.response)
+            if not error.response['Error']['Code'] == 'AlreadyExists':
+                raise
+
+    def add_scaling_group_to_jenkins(self):
+        print(f"{self.region.region_name}: add_scaling_group_to_jenkins")
```
Comment (vponomaryov) [inline sdcm/utils/aws_builder.py:313]: and here

File: vars/getJenkinsLabels.groovy [hunk 10]
```diff
@@ -35,7 +35,7 @@ def call(String backend, String region=null, String datacenter=null, String loca
                           'gce-us-east1': "${gcp_project}-builders-us-east1",
                           'gce-us-west1': "${gcp_project}-builders-us-west1",
                           'gce': "${gcp_project}-builders",
-                          'docker': 'sct-builders',
+                          'aws': 'aws-sct-builders-eu-west-1-v2-asg',
```
[the remark this answers is not in the document]
Comment (fruch) [reply vars/getJenkinsLabels.groovy:38]: But getCloudProviderFromBackend is used across the pipelines for other purposes It is not supposed to return anything else but aws/gcp/azure...

File: sdcm/utils/aws_builder.py [hunk 11]
```diff
@@ -137,18 +209,144 @@ def associate_elastic_ip(self):
...
 
-    def configure(self):
+    def configure_one_builder(self):
         self.region.create_sct_ssh_security_group()
         self.create_elastic_ip()
         self.associate_elastic_ip()
         self.add_to_jenkins()
 
+    def create_launch_template(self):
+        print(f"{self.region.region_name}: create_launch_template")
```
[the remark this answers is not in the document]
Comment (fruch) [reply sdcm/utils/aws_builder.py:219]: Can switch to logger, but the hydra commands using that weren't configuring logging at all.

It's not gonna run anywhere but only manually when reconfiguring those parts, or adding a new region.

[the remark this answers is not in the document]
Comment (vponomaryov) [reply sdcm/utils/aws_builder.py:219]: I asked  because exactly `LOGGER.debug` gets used below.
So, as of now it is mix, and it doesn't look ok to me.

File: sdcm/utils/aws_builder.py (hunk 6 above)
Comment (roydahan) [reply sdcm/utils/aws_builder.py]: Let's start with 4 and see how it works.

File: vars/getJenkinsLabels.groovy (hunk 10 above)
[the remark this answers is not in the document]
Comment (fruch) [reply vars/getJenkinsLabels.groovy:30]: `getCloudProviderFromBackend` returns cloud providers (and used in multiple places, not just here), `aws-eu-west-1` isn't cloud provider

File: defaults/aws_config.yaml [hunk 12]
```diff
@@ -9,23 +9,23 @@ instance_type_monitor: 't3.large'
...
+    ami_id_monitor: 'ami-06878d265978313ca' # ubuntu/images/hvm-ssd/ubuntu-jammy-22.04-amd64-server-20221206 Canonical, Ubuntu, 22.04 LTS, amd64 jammy image build on 2022-12-06
   us-west-2: # US West (Oregon)
-    ami_id_loader: 'ami-04d2eb44d523b3ccb' # Loader dedicated AMI v19
-    ami_id_monitor: 'ami-01ed306a12b7d1c96' # Official CentOS Linux 7 x86_64 HVM EBS ENA 1901_01
-  eu-west-1: # Europe (Ireland)
-    ami_id_loader: 'ami-042cf1bc21e30ce60' # Loader dedicated AMI v19
-    ami_id_monitor: 'ami-0ff760d16d9497662' # Official CentOS Linux 7 x86_64 HVM EBS ENA 1901_01
+    ami_id_loader: 'ami-09808043b7fcdb244' # Loader dedicated AMI scylla-qa-loader-ami-v19
+    ami_id_monitor: 'ami-03f8756d29f0b5f21' # ubuntu/images/hvm-ssd/ubuntu-jammy-22.04-amd64-server-20221206 Canonical, Ubuntu, 22.04 LTS, amd64 jammy image build on 2022-12-06
+  eu-west-1: # EU (Ireland)
```
Comment (vponomaryov) [inline defaults/aws_config.yaml:17]: Here `Europe` is changed to `EU`, but lines `20` and `23` stay with `Europe`.
So, need either to revert this change back or update those.
Personally, I don't see a need to change it.

File: vars/byoLongevityPipeline.groovy [hunk 13]
```diff
@@ -39,7 +39,7 @@ def call() {
                description: 'aws|gce',
                name: 'backend')
             string(defaultValue: "eu-west-1",
-               description: 'Supported: us-east-1|eu-west-1|eu-west-2|eu-north-1|random (randomly select region)',
+               description: 'Supported: us-east-1 | eu-west-1 | eu-west-2 | eu-north-1 | eu-central-1 | random (randomly select region)',
```
Comment (vponomaryov) [inline vars/byoLongevityPipeline.groovy:42]: `us-west-2` is missing

File: vars/cdcReplicationPipeline.groovy [hunk 14]
```diff
@@ -89,7 +89,7 @@ def call(Map pipelineParams) {
                name: 'backend')
 
             string(defaultValue: "${pipelineParams.get('region', 'eu-west-1')}",
-               description: 'Supported: us-east-1|eu-west-1|eu-west-2|eu-north-1|random (randomly select region)',
+               description: 'Supported: us-east-1 | eu-west-1 | eu-west-2 | eu-north-1 | eu-central-1 | random (randomly select region)',
```
Comment (vponomaryov) [inline vars/cdcReplicationPipeline.groovy:92]: same here

File: vars/getJenkinsLabels.groovy [hunk 15]
```diff
@@ -35,7 +36,7 @@ def call(String backend, String region=null, String datacenter=null, String loca
         def supported_regions = []
 
         if (cloud_provider == 'aws') {
-            supported_regions = ["eu-west-2", "eu-north-1", "eu-central-1"]
+            supported_regions = ["eu-west-2", "eu-north-1", "eu-central-1", "us-west-2"]
```
Comment (vponomaryov) [inline vars/getJenkinsLabels.groovy:39]: We support 6 regions, why here we have just 4?

File: vars/longevityPipeline.groovy [hunk 16]
```diff
@@ -26,7 +26,7 @@ def call(Map pipelineParams) {
                name: 'backend')
 
             string(defaultValue: "${pipelineParams.get('region', 'eu-west-1')}",
-               description: 'Supported: us-east-1|eu-west-1|eu-west-2|eu-north-1|random (randomly select region)',
+               description: 'Supported: us-east-1 | eu-west-1 | eu-west-2 | eu-north-1 | eu-central-1 | random (randomly select region)',
```
Comment (vponomaryov) [inline vars/longevityPipeline.groovy:29]: Same here.

File: vars/managerPipeline.groovy [hunk 17]
```diff
@@ -58,7 +58,7 @@ def call(Map pipelineParams) {
                description: 'aws|gce',
                name: 'backend')
             string(defaultValue: "${pipelineParams.get('region', 'eu-west-1')}",
-               description: 'Supported: us-east-1|eu-west-1|eu-west-2|eu-north-1|random (randomly select region)',
+               description: 'Supported: us-east-1 | eu-west-1 | eu-west-2 | eu-north-1 | eu-central-1 | random (randomly select region)',
```
Comment (vponomaryov) [inline vars/managerPipeline.groovy:61]: Same here

File: vars/rollingOperatorUpgradePipeline.groovy [hunk 18]
```diff
@@ -17,7 +17,7 @@ def call(Map pipelineParams) {
             choice(choices: ["${pipelineParams.get('backend', 'k8s-gke')}", 'k8s-eks', 'k8s-gke'],
                    name: 'backend')
             string(defaultValue: "${pipelineParams.get('region', '')}",
-               description: 'Supported: us-east-1|eu-west-1|eu-west-2|eu-north-1|random (randomly select region)',
+               description: 'Supported: us-east-1 | eu-west-1 | eu-west-2 | eu-north-1 | eu-central-1 | random (randomly select region)',
```
Comment (vponomaryov) [inline vars/rollingOperatorUpgradePipeline.groovy:20]: Same here

File: vars/rollingUpgradePipeline.groovy [hunk 19]
```diff
@@ -26,7 +26,7 @@ def call(Map pipelineParams) {
                name: 'backend')
 
             string(defaultValue: "${pipelineParams.get('region', 'eu-west-1')}",
-               description: 'Supported: us-east-1|eu-west-1|eu-west-2|eu-north-1|random (randomly select region)',
+               description: 'Supported: us-east-1 | eu-west-1 | eu-west-2 | eu-north-1 | eu-central-1 | random (randomly select region)',
```
Comment (vponomaryov) [inline vars/rollingUpgradePipeline.groovy:29]: Same here

File: utils/copy_ami_to_all_regions.sh [hunk 20]
```diff
@@ -0,0 +1,10 @@
+AMI_ID=ami-00c1a16b5136fbb7c
+AMI_NAME='scylla-qa-loader-ami-v19'
+SOURCE_REGION=us-east-1
+TARGET_REGIONS='us-west-2 eu-west-1 eu-west-1 eu-west-2 eu-central-1'
```
Comment (vponomaryov) [inline utils/copy_ami_to_all_regions.sh:4]: The `eu-west-1` one is duplicated twice and `eu-north-1` is missing.

File: utils/create_aws_region_data.py [hunk 21]
```diff
@@ -0,0 +1,60 @@
...
+    'eu-west-3': 'Europe (Paris)',
+    'eu-north-1': 'Europe (Stockholm)',
+    'me-south-1': 'Middle East (Bahrain)',
+    'sa-east-1': 'South America (São Paulo)',
+}
+
+
+def main():
+
+    for region in ['us-east-1', 'us-west-2', 'eu-west-1', 'eu-west-1', 'eu-west-2', 'eu-central-1']:
```
Comment (vponomaryov) [inline utils/create_aws_region_data.py:32]: Same here. Duplicated `eu-west-1` and missing `eu-north-1`.

File: unit_tests/test_base_version.py [hunk 22]
```diff
@@ -33,7 +33,7 @@ def test_master(self):
         scylla_repo = self.url_base + 'unstable/scylla/master/rpm/centos/2021-08-29T00:58:58Z/scylla.repo'
         linux_distro = 'centos'
         version_list = general_test(scylla_repo, linux_distro)
-        self.assertEqual(version_list, ['5.1'])
+        self.assertEqual(version_list, ['5.2'])
```
Comment (vponomaryov) [inline unit_tests/test_base_version.py:36]: I guess this commit must be removed because it is already part of another PR

**Review by vponomaryov — CHANGES_REQUESTED** [review]

File: defaults/aws_config.yaml (hunk 12 above)
Comment (fruch) [reply defaults/aws_config.yaml:17]: It's a change in comment, cause of the code I've taken from siren-tests.
nitpicking from my POV.

File: vars/byoLongevityPipeline.groovy (hunk 13 above)
Comment (fruch) [reply vars/byoLongevityPipeline.groovy]: fixed

File: vars/cdcReplicationPipeline.groovy (hunk 14 above)
Comment (fruch) [reply vars/cdcReplicationPipeline.groovy]: fixed

File: vars/rollingUpgradePipeline.groovy (hunk 19 above)
Comment (fruch) [reply vars/rollingUpgradePipeline.groovy]: fixed

File: utils/create_aws_region_data.py (hunk 21 above)
Comment (fruch) [reply utils/create_aws_region_data.py]: fixed

File: unit_tests/test_base_version.py (hunk 22 above)
Comment (fruch) [reply unit_tests/test_base_version.py]: fixed

File: vars/rollingOperatorUpgradePipeline.groovy (hunk 18 above)
Comment (fruch) [reply vars/rollingOperatorUpgradePipeline.groovy]: fixed

File: utils/copy_ami_to_all_regions.sh (hunk 20 above)
Comment (fruch) [reply utils/copy_ami_to_all_regions.sh]: fixed

File: vars/longevityPipeline.groovy (hunk 16 above)
Comment (fruch) [reply vars/longevityPipeline.groovy]: fixed

File: defaults/aws_config.yaml (hunk 12 above)
Comment (vponomaryov) [reply defaults/aws_config.yaml:17]: It is totally normal to reduce the number of changes in a huge PR.

File: sdcm/utils/aws_builder.py [hunk 23]
```diff
@@ -137,18 +208,144 @@ def associate_elastic_ip(self):
...
 
-    def configure(self):
+    def configure_one_builder(self):
         self.region.create_sct_ssh_security_group()
         self.create_elastic_ip()
         self.associate_elastic_ip()
         self.add_to_jenkins()
 
+    def create_launch_template(self):
+        LOGGER.info(f"{self.region.region_name}: create_launch_template")
```
Comment (vponomaryov) [inline sdcm/utils/aws_builder.py:218]: It makes the precommit fail.
Need to use lazy formatting in the logging lines.

See: https://jenkins.scylladb.com/job/sct-github-PRs-scan/job/scylla-cluster-tests/job/PR-5585/58/execution/node/23/log/

File: sdcm/utils/aws_builder.py [hunk 24]
```diff
@@ -137,18 +208,144 @@ def associate_elastic_ip(self):
...
+                    },
+                ],
+            )
+        except botocore.exceptions.ClientError as error:
+            LOGGER.debug(error.response)
+            if not error.response['Error']['Code'] == 'InvalidLaunchTemplateName.AlreadyExistsException':
+                raise
+
+    def create_auto_scaling_group(self):
+        LOGGER.info(f"{self.region.region_name}: create_auto_scaling_group")
```
Comment (vponomaryov) [inline sdcm/utils/aws_builder.py:248]: Same here.

File: sdcm/utils/aws_builder.py [hunk 25]
```diff
@@ -137,18 +208,144 @@ def associate_elastic_ip(self):
...
+                                                     }]
+                                                 )
+
+        except botocore.exceptions.ClientError as error:
+            LOGGER.debug(error.response)
+            if not error.response['Error']['Code'] == 'AlreadyExists':
+                raise
+
+    def add_scaling_group_to_jenkins(self):
+        LOGGER.info(f"{self.region.region_name}: add_scaling_group_to_jenkins")
```
Comment (vponomaryov) [inline sdcm/utils/aws_builder.py:312]: Same here.

File: vars/getJenkinsLabels.groovy (hunk 15 above)
Comment (vponomaryov) [reply vars/getJenkinsLabels.groovy:39]: O.k.

**Review by vponomaryov — CHANGES_REQUESTED** [review]

File: sdcm/utils/aws_builder.py (hunk 23 above)
Comment (fruch) [reply sdcm/utils/aws_builder.py]: fixed

File: sdcm/utils/aws_builder.py (hunk 25 above)
Comment (fruch) [reply sdcm/utils/aws_builder.py]: fixed

File: sdcm/utils/aws_builder.py (hunk 24 above)
Comment (fruch) [reply sdcm/utils/aws_builder.py]: fixed

**Review by roydahan — APPROVED** [review]

**Comment by fruch** [comment]
@roydahan we might not want this in branch 5.2

**Comment by roydahan** [comment]
why not?

**Comment by fruch** [comment]
> why not?

It might introduce noise we aren't expecting, 

Let's merge, and let it work on master, and decide
